iceoryx_hoofs/include/iceoryx_hoofs/cxx/function_ref.hpp

```diff
@@ -87,10 +87,17 @@ class function_ref<ReturnType(ArgTypes...)>
     /// @brief Creates a function_ref with a callable whose lifetime has to be longer than function_ref
     /// @param[in] callable that is not a function_ref
     template <typename CallableType,
-              typename = std::enable_if_t<!is_function_ref<std::remove_reference_t<CallableType>>::value>,
-              typename = std::enable_if_t<is_invocable<CallableType, ArgTypes...>::value>>
+              typename = std::enable_if_t<!is_function_pointer<CallableType>::value
+                                          && not_same<CallableType, function_ref>::value
```

Comment:
They do not, a function_ref is not a function pointer. I do not think the second conition can be removed without changing the meaning. Also the function_ref test was there before my modification, I just needed to exclude function pointers here because the ctor was not working correctly with them and by excluding them overload resolution is guided to the explicit ctor for function pointers (it would not without this).